core/server/proxy/src/main/java/alluxio/proxy/s3/S3ObjectTask.java

```diff
@@ -483,15 +469,19 @@ public Response continueTask() {
               .setMode(PMode.newBuilder()
                   .setOwnerBits(Bits.ALL)
                   .setGroupBits(Bits.ALL)
-                  .setOtherBits(Bits.NONE).build());
+                  .setOtherBits(Bits.NONE)
+                  .build())
+              .setWriteType(S3RestUtils.getS3WriteType())
+              .setXattrPropStrat(XAttrPropagationStrategy.LEAF_NODE)
+              .setOverwrite(true);
```

Comment:
Do we need the `.putAllXattr(xattrMap)` here? I see we have it in the CreateFilePOptions.

Reply:
its actually pick and choosing the needed ones from xattrMap to put into this createfoption for copy, as in some attrs we are getting from src file, some we r getting the user-passed-in fields, depends on the x-amz-metadata-directive header